### PR TITLE
add utility functions for group requests

### PR DESCRIPTION
**Pull request by saweber**
This is for powering group requests, and adds the benefit of slightly decoupling utility services from request handlers.

## What does this PR change?
* Adds utility functions and structs for Allocation filters and CloudCost filters

## Does this PR relate to any other PRs?
* yes, https://github.com/kubecost/kubecost-cost-model/pull/1478/files

## How will this PR impact users?
* no effect for opencost, enables new kubecost functionality

## Does this PR address any GitHub or Zendesk issues?
* CORE-234

## How was this PR tested?
* existing + new automated tests

## Does this PR require changes to documentation?
* no

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* no permissions

### CODE REVIEW

File: pkg/filter/util/cloudcost.go [hunk 1]
```diff
@@ -53,6 +53,61 @@ func CloudCostFilterFromParams(pmr mapper.PrimitiveMapReader) filter.Filter[*kub
...
+	FilterCategories       []string `json:"filterCategories"`
+	FilterLabels           []string `json:"filterLabels"`
+}
+
+func (g *CloudCostFilters) Equals(that CloudCostFilters) bool {
+	// FIXME
+	return false
+}
+
+func ParseCloudCostFilter(filters CloudCostFilters) filter.Filter[*kubecost.CloudCost] {
```
Comment (saweber) [inline pkg/filter/util/cloudcost.go:71]: this function is copy / paste. I plan to refactor the existing implementation into a transform and use this for the logic, to remove duplication.

File: pkg/util/allocationfilterutil/queryfilters.go [hunk 2]
```diff
@@ -416,3 +416,220 @@ func filterV1DoubleValueFromList(rawFilterValuesUnsplit []string, filterField ku
...
+	FilterEnvironments    []string `json:"filterEnvironments"`
+	FilterOwners          []string `json:"filterOwners"`
+	FilterProducts        []string `json:"filterProducts"`
+	FilterTeams           []string `json:"filterTeams"`
+	FilterAnnotations     []string `json:"filterAnnotations"`
+	FilterLabels          []string `json:"filterLabels"`
+	FilterServices        []string `json:"filterServices"`
+}
+
+func ParseAllocationFilterV1(filters FilterV1, labelConfig *kubecost.LabelConfig, clusterMap clusters.ClusterMap) kubecost.AllocationFilter {
```
Comment (saweber) [inline pkg/util/allocationfilterutil/queryfilters.go:438]: this function is copy / paste. I plan to refactor the existing implementation into a transform and use this for the logic, to remove duplication.

File: pkg/filter/util/cloudcost.go [hunk 3]
```diff
@@ -53,6 +53,61 @@ func CloudCostFilterFromParams(pmr mapper.PrimitiveMapReader) filter.Filter[*kub
...
+	FilterAccountIDs       []string `json:"filterAccountIDs"`
+	FilterProviders        []string `json:"filterProviders"`
+	FilterProviderIDs      []string `json:"filterProviderIDs"`
+	FilterServices         []string `json:"filterServices"`
+	FilterCategories       []string `json:"filterCategories"`
+	FilterLabels           []string `json:"filterLabels"`
+}
+
+func (g *CloudCostFilters) Equals(that CloudCostFilters) bool {
+	// FIXME
```
Comment (saweber) [inline pkg/filter/util/cloudcost.go:67]: will fix as part of this

File: pkg/util/allocationfilterutil/queryfilters.go [hunk 4]
```diff
@@ -416,3 +416,220 @@ func filterV1DoubleValueFromList(rawFilterValuesUnsplit []string, filterField ku
...
+	FilterContainers      []string `json:"filterContainers"`
+	FilterDepartments     []string `json:"filterDepartments"`
+	FilterEnvironments    []string `json:"filterEnvironments"`
+	FilterOwners          []string `json:"filterOwners"`
+	FilterProducts        []string `json:"filterProducts"`
+	FilterTeams           []string `json:"filterTeams"`
+	FilterAnnotations     []string `json:"filterAnnotations"`
+	FilterLabels          []string `json:"filterLabels"`
+	FilterServices        []string `json:"filterServices"`
+}
```
Comment (saweber) [inline pkg/util/allocationfilterutil/queryfilters.go:436]: missing equals() function

**Review by Sean-Holcomb — APPROVED** [review]
Looks good, good unit tests

**Pull request merged**